app/code/Magento/CacheInvalidate/Model/PurgeCache.php

```diff
@@ -131,10 +132,19 @@ private function sendPurgeRequestToServers($socketAdapter, $servers, $formattedT
                 $socketAdapter->read();
                 $socketAdapter->close();
             } catch (\Exception $e) {
-                $this->logger->critical($e->getMessage(), compact('server', 'formattedTagsChunk'));
-                return false;
+                $unresponsiveServerCount++;
+                if ($unresponsiveServerCount == count($servers)) {
```

Comment:
I propose to keep accumulating information in the catch body, however, it might be better to move the logging after the foreach to reduce the amount of noise in logs and to log the single final message.

Also, I think it will be very nice to include the unresponsive servers' information into the message (like the list of unresponsive servers hosts)

Reply:
Yup, I agree. I've created a new commit. Good thought :)